{
    "active_lock_reason": null,
    "assignee": null,
    "assignees": [],
    "author_association": "NONE",
    "body": "### Describe the bug\r\n\r\nThe included `Cython/Includes/numpy/__init__.pxd` is not compatible with NumPy 2.0. In particular, the `PyDataType_SHAPE` [helper][1] generates code\r\n\r\n```\r\n    /* \"Cython/Includes/numpy/__init__.pxd\":796\r\n * cdef inline tuple PyDataType_SHAPE(dtype d):\r\n *     if PyDataType_HASSUBARRAY(d):\r\n *         return <tuple>d.subarray.shape             # <<<<<<<<<<<<<<\r\n *     else:\r\n *         return ()\r\n */\r\n    __Pyx_XDECREF(__pyx_r);\r\n    __Pyx_INCREF(((PyObject*)__pyx_v_d->subarray->shape));\r\n    __pyx_r = ((PyObject*)__pyx_v_d->subarray->shape);\r\n    goto __pyx_L0;\r\n```\r\n\r\nThis causes compiler errors because `PyArray_Descr` is [now more opaque][2]\r\n\r\n```\r\n...\r\n      src/python/bezier/_speedup.c:20608:39: error: ‘PyArray_Descr’ {aka ‘struct _PyArray_Descr’} has no member named ‘subarray’\r\n      20608 |     __Pyx_INCREF(((PyObject*)__pyx_v_d->subarray->shape));\r\n            |                                       ^~\r\n...\r\n      src/python/bezier/_speedup.c:20609:36: error: ‘PyArray_Descr’ {aka ‘struct _PyArray_Descr’} has no member named ‘subarray’\r\n      20609 |     __pyx_r = ((PyObject*)__pyx_v_d->subarray->shape);\r\n            |                                    ^~\r\n```\r\n\r\nSee https://github.com/dhermes/bezier/pull/319 (including a hacky workaround)\r\n\r\n[1]: https://github.com/cython/cython/blob/3.0.10/Cython/Includes/numpy/__init__.pxd#L794-L798\r\n[2]: https://numpy.org/doc/stable/numpy_2_0_migration_guide.html#the-pyarray-descr-struct-has-been-changed\r\n\r\n### Code to reproduce the behaviour:\r\n\r\nAnything that imports NumPy\r\n\r\n### Expected behaviour\r\n\r\nClean build\r\n\r\n### OS\r\n\r\nLinux, macOS, Windows\r\n\r\n### Python version\r\n\r\n3.11, 3.12\r\n\r\n### Cython version\r\n\r\n3.0.10\r\n\r\n### Additional context\r\n\r\n_No response_",
    "closed_at": null,
    "closed_by": null,
    "comment_data": [
        {
            "author_association": "MEMBER",
            "body": "We don't maintain the numpy pxd files that are bundled with Cython any more. We've handed off the maintenance to the Numpy project itself.\r\n\r\nWhat *should* be happening is that it should be finding the equivalent .pxd files that are bundled with Numpy and using those first. From what you've posted here, I'm not quite sure why that isn't happening for you.",
            "created_at": "2024-06-18T17:37:48Z",
            "html_url": "https://github.com/cython/cython/issues/6249#issuecomment-2176633822",
            "id": 2176633822,
            "issue_url": "https://api.github.com/repos/cython/cython/issues/6249",
            "node_id": "IC_kwDOABDGAc6BvMve",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/cython/cython/issues/comments/2176633822/reactions"
            },
            "updated_at": "2024-06-18T17:37:48Z",
            "url": "https://api.github.com/repos/cython/cython/issues/comments/2176633822",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/10536947?v=4",
                "events_url": "https://api.github.com/users/da-woods/events{/privacy}",
                "followers_url": "https://api.github.com/users/da-woods/followers",
                "following_url": "https://api.github.com/users/da-woods/following{/other_user}",
                "gists_url": "https://api.github.com/users/da-woods/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/da-woods",
                "id": 10536947,
                "login": "da-woods",
                "node_id": "MDQ6VXNlcjEwNTM2OTQ3",
                "organizations_url": "https://api.github.com/users/da-woods/orgs",
                "received_events_url": "https://api.github.com/users/da-woods/received_events",
                "repos_url": "https://api.github.com/users/da-woods/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/da-woods/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/da-woods/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/da-woods",
                "user_view_type": "public"
            }
        },
        {
            "author_association": "MEMBER",
            "body": "The aim is to remove the files from Cython reasonably soon, but it won't be in a 3.0.x release because those are meant to be bugfix-only releases.",
            "created_at": "2024-06-18T17:38:44Z",
            "html_url": "https://github.com/cython/cython/issues/6249#issuecomment-2176635292",
            "id": 2176635292,
            "issue_url": "https://api.github.com/repos/cython/cython/issues/6249",
            "node_id": "IC_kwDOABDGAc6BvNGc",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/cython/cython/issues/comments/2176635292/reactions"
            },
            "updated_at": "2024-06-18T17:38:44Z",
            "url": "https://api.github.com/repos/cython/cython/issues/comments/2176635292",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/10536947?v=4",
                "events_url": "https://api.github.com/users/da-woods/events{/privacy}",
                "followers_url": "https://api.github.com/users/da-woods/followers",
                "following_url": "https://api.github.com/users/da-woods/following{/other_user}",
                "gists_url": "https://api.github.com/users/da-woods/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/da-woods",
                "id": 10536947,
                "login": "da-woods",
                "node_id": "MDQ6VXNlcjEwNTM2OTQ3",
                "organizations_url": "https://api.github.com/users/da-woods/orgs",
                "received_events_url": "https://api.github.com/users/da-woods/received_events",
                "repos_url": "https://api.github.com/users/da-woods/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/da-woods/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/da-woods/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/da-woods",
                "user_view_type": "public"
            }
        },
        {
            "author_association": "NONE",
            "body": "@da-woods totally hear you, but as it stands, the `3.0.10` release is producing broken C code.\r\n\r\nIn particular the `cython` CLI is adding in code based on the `Cython/Includes/numpy/__init__.pxd` file even though I haven't explicitly asked for it. The `_speedup.c` file in https://github.com/dhermes/bezier is produced with no flags at all:\r\n\r\n```\r\ncython src/python/bezier/_speedup.pyx\r\n```\r\n\r\nIf there is a way to tell the `cython` CLI to **exclude** `Cython/Includes/numpy/`, I'd happily use that.\r\n\r\nGiven the removal of `numpy/__init__.pxd` in https://github.com/cython/cython/pull/5842, is there any reason why you wouldn't do a `3.1.x` release?\r\n\r\n----\r\n\r\nI.e. is there a straightforward way I could control:\r\n\r\n```c\r\n/* #### Code section: filename_table ### */\r\n\r\nstatic const char *__pyx_f[] = {\r\n  \"src/python/bezier/_speedup.pyx\",\r\n  \"<stringsource>\",\r\n  \"Cython/Includes/numpy/__init__.pxd\",\r\n  \"Cython/Includes/cpython/type.pxd\",\r\n};\r\n```",
            "created_at": "2024-06-18T18:37:48Z",
            "html_url": "https://github.com/cython/cython/issues/6249#issuecomment-2176730104",
            "id": 2176730104,
            "issue_url": "https://api.github.com/repos/cython/cython/issues/6249",
            "node_id": "IC_kwDOABDGAc6BvkP4",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 2,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 2,
                "url": "https://api.github.com/repos/cython/cython/issues/comments/2176730104/reactions"
            },
            "updated_at": "2024-06-18T18:39:05Z",
            "url": "https://api.github.com/repos/cython/cython/issues/comments/2176730104",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/520669?v=4",
                "events_url": "https://api.github.com/users/dhermes/events{/privacy}",
                "followers_url": "https://api.github.com/users/dhermes/followers",
                "following_url": "https://api.github.com/users/dhermes/following{/other_user}",
                "gists_url": "https://api.github.com/users/dhermes/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/dhermes",
                "id": 520669,
                "login": "dhermes",
                "node_id": "MDQ6VXNlcjUyMDY2OQ==",
                "organizations_url": "https://api.github.com/users/dhermes/orgs",
                "received_events_url": "https://api.github.com/users/dhermes/received_events",
                "repos_url": "https://api.github.com/users/dhermes/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/dhermes/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/dhermes/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/dhermes",
                "user_view_type": "public"
            }
        },
        {
            "author_association": "MEMBER",
            "body": "Do you have the correct NumPy version installed (i.e. importable) while calling `cython` in your build?",
            "created_at": "2024-06-18T18:46:00Z",
            "html_url": "https://github.com/cython/cython/issues/6249#issuecomment-2176742455",
            "id": 2176742455,
            "issue_url": "https://api.github.com/repos/cython/cython/issues/6249",
            "node_id": "IC_kwDOABDGAc6BvnQ3",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/cython/cython/issues/comments/2176742455/reactions"
            },
            "updated_at": "2024-06-18T18:46:00Z",
            "url": "https://api.github.com/repos/cython/cython/issues/comments/2176742455",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/491659?v=4",
                "events_url": "https://api.github.com/users/scoder/events{/privacy}",
                "followers_url": "https://api.github.com/users/scoder/followers",
                "following_url": "https://api.github.com/users/scoder/following{/other_user}",
                "gists_url": "https://api.github.com/users/scoder/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/scoder",
                "id": 491659,
                "login": "scoder",
                "node_id": "MDQ6VXNlcjQ5MTY1OQ==",
                "organizations_url": "https://api.github.com/users/scoder/orgs",
                "received_events_url": "https://api.github.com/users/scoder/received_events",
                "repos_url": "https://api.github.com/users/scoder/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/scoder/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/scoder/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/scoder",
                "user_view_type": "public"
            }
        },
        {
            "author_association": "MEMBER",
            "body": "> is there any reason why you wouldn't do a `3.1.x` release?\r\n\r\nIt's not ready yet.",
            "created_at": "2024-06-18T18:47:06Z",
            "html_url": "https://github.com/cython/cython/issues/6249#issuecomment-2176744107",
            "id": 2176744107,
            "issue_url": "https://api.github.com/repos/cython/cython/issues/6249",
            "node_id": "IC_kwDOABDGAc6Bvnqr",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/cython/cython/issues/comments/2176744107/reactions"
            },
            "updated_at": "2024-06-18T18:47:06Z",
            "url": "https://api.github.com/repos/cython/cython/issues/comments/2176744107",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/491659?v=4",
                "events_url": "https://api.github.com/users/scoder/events{/privacy}",
                "followers_url": "https://api.github.com/users/scoder/followers",
                "following_url": "https://api.github.com/users/scoder/following{/other_user}",
                "gists_url": "https://api.github.com/users/scoder/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/scoder",
                "id": 491659,
                "login": "scoder",
                "node_id": "MDQ6VXNlcjQ5MTY1OQ==",
                "organizations_url": "https://api.github.com/users/scoder/orgs",
                "received_events_url": "https://api.github.com/users/scoder/received_events",
                "repos_url": "https://api.github.com/users/scoder/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/scoder/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/scoder/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/scoder",
                "user_view_type": "public"
            }
        },
        {
            "author_association": "NONE",
            "body": "> Do you have the correct NumPy version installed (i.e. importable) while calling `cython` in your build?\r\n\r\nI do not! Using `nox` I made a virtualenv with only `Cython==3.0.10` as the dependency.\r\n\r\nAdding `numpy==2.0.0` to that virtualenv fixed the issue, thanks!\r\n\r\n----\r\n\r\nHopefully this issue will serve as documentation / something people can search for on this issue.\r\n\r\nI have been updating my `_speedup.c` with this virtual environment since August 2017 or so and have never installed `numpy` into it. I wonder how many others are in this same situation (seems like not too many given the Google search results / GitHub issue search results on this topic).",
            "created_at": "2024-06-18T20:28:13Z",
            "html_url": "https://github.com/cython/cython/issues/6249#issuecomment-2176913963",
            "id": 2176913963,
            "issue_url": "https://api.github.com/repos/cython/cython/issues/6249",
            "node_id": "IC_kwDOABDGAc6BwRIr",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 2,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 2,
                "url": "https://api.github.com/repos/cython/cython/issues/comments/2176913963/reactions"
            },
            "updated_at": "2024-06-18T20:28:13Z",
            "url": "https://api.github.com/repos/cython/cython/issues/comments/2176913963",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/520669?v=4",
                "events_url": "https://api.github.com/users/dhermes/events{/privacy}",
                "followers_url": "https://api.github.com/users/dhermes/followers",
                "following_url": "https://api.github.com/users/dhermes/following{/other_user}",
                "gists_url": "https://api.github.com/users/dhermes/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/dhermes",
                "id": 520669,
                "login": "dhermes",
                "node_id": "MDQ6VXNlcjUyMDY2OQ==",
                "organizations_url": "https://api.github.com/users/dhermes/orgs",
                "received_events_url": "https://api.github.com/users/dhermes/received_events",
                "repos_url": "https://api.github.com/users/dhermes/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/dhermes/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/dhermes/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/dhermes",
                "user_view_type": "public"
            }
        },
        {
            "author_association": "NONE",
            "body": "Just in case it's useful to anyone, I had the same error, but it was fixed by by making sure to do, e.g.,\r\n\r\n```\r\ncimport numpy as cnp\r\n\r\ncnp.import_array()  # add this line\r\n```\r\n\r\nas described [here](https://cython.readthedocs.io/en/latest/src/tutorial/numpy.html#adding-types).",
            "created_at": "2024-07-03T10:26:58Z",
            "html_url": "https://github.com/cython/cython/issues/6249#issuecomment-2205735439",
            "id": 2205735439,
            "issue_url": "https://api.github.com/repos/cython/cython/issues/6249",
            "node_id": "IC_kwDOABDGAc6DeNoP",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/cython/cython/issues/comments/2205735439/reactions"
            },
            "updated_at": "2024-07-03T10:26:58Z",
            "url": "https://api.github.com/repos/cython/cython/issues/comments/2205735439",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/1421092?v=4",
                "events_url": "https://api.github.com/users/mattpitkin/events{/privacy}",
                "followers_url": "https://api.github.com/users/mattpitkin/followers",
                "following_url": "https://api.github.com/users/mattpitkin/following{/other_user}",
                "gists_url": "https://api.github.com/users/mattpitkin/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/mattpitkin",
                "id": 1421092,
                "login": "mattpitkin",
                "node_id": "MDQ6VXNlcjE0MjEwOTI=",
                "organizations_url": "https://api.github.com/users/mattpitkin/orgs",
                "received_events_url": "https://api.github.com/users/mattpitkin/received_events",
                "repos_url": "https://api.github.com/users/mattpitkin/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/mattpitkin/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/mattpitkin/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/mattpitkin",
                "user_view_type": "public"
            }
        },
        {
            "author_association": "NONE",
            "body": "> Just in case it's useful to anyone, I had the same error, but it was fixed by by making sure to do, e.g.,\r\n> \r\n> ```\r\n> cimport numpy as cnp\r\n> \r\n> cnp.import_array()  # add this line\r\n> ```\r\n> \r\n> as described [here](https://cython.readthedocs.io/en/latest/src/tutorial/numpy.html#adding-types).\r\n\r\nDid you change something else? I followed your edit but still get the same error with **PyArray_Descr** when installing [BioExt](https://github.com/veg/BioExt)\r\n\r\n```\r\n      BioExt/align/_align.c:18741:41: error: no member named 'subarray' in 'struct _PyArray_Descr'\r\n          __Pyx_INCREF(((PyObject*)__pyx_v_d->subarray->shape));\r\n                                   ~~~~~~~~~  ^\r\n      BioExt/align/_align.c:2065:37: note: expanded from macro '__Pyx_INCREF'\r\n        #define __Pyx_INCREF(r) Py_INCREF(r)\r\n                                          ^\r\n      /Users/chjulian/opt/miniconda3/include/python3.9/object.h:411:49: note: expanded from macro 'Py_INCREF'\r\n      #define Py_INCREF(op) _Py_INCREF(_PyObject_CAST(op))\r\n                                                      ^~\r\n      /Users/chjulian/opt/miniconda3/include/python3.9/object.h:112:41: note: expanded from macro '_PyObject_CAST'\r\n      #define _PyObject_CAST(op) ((PyObject*)(op))\r\n                                              ^~\r\n      BioExt/align/_align.c:18742:38: error: no member named 'subarray' in 'struct _PyArray_Descr'\r\n          __pyx_r = ((PyObject*)__pyx_v_d->subarray->shape);\r\n                                ~~~~~~~~~  ^\r\n      2 errors generated.\r\n      error: command '/usr/bin/clang' failed with exit code 1\r\n\r\n```",
            "created_at": "2024-07-09T14:39:32Z",
            "html_url": "https://github.com/cython/cython/issues/6249#issuecomment-2217911448",
            "id": 2217911448,
            "issue_url": "https://api.github.com/repos/cython/cython/issues/6249",
            "node_id": "IC_kwDOABDGAc6EMqSY",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/cython/cython/issues/comments/2217911448/reactions"
            },
            "updated_at": "2024-07-09T14:39:32Z",
            "url": "https://api.github.com/repos/cython/cython/issues/comments/2217911448",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/7526558?v=4",
                "events_url": "https://api.github.com/users/Chjulian/events{/privacy}",
                "followers_url": "https://api.github.com/users/Chjulian/followers",
                "following_url": "https://api.github.com/users/Chjulian/following{/other_user}",
                "gists_url": "https://api.github.com/users/Chjulian/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/Chjulian",
                "id": 7526558,
                "login": "Chjulian",
                "node_id": "MDQ6VXNlcjc1MjY1NTg=",
                "organizations_url": "https://api.github.com/users/Chjulian/orgs",
                "received_events_url": "https://api.github.com/users/Chjulian/received_events",
                "repos_url": "https://api.github.com/users/Chjulian/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/Chjulian/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/Chjulian/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/Chjulian",
                "user_view_type": "public"
            }
        },
        {
            "author_association": "MEMBER",
            "body": "I don't see why `import_array` should affect the paths used to search for files so I  don't think it will be a solution here.\r\n\r\nI don't think I fully understand why this problem is happening though",
            "created_at": "2024-07-09T16:56:20Z",
            "html_url": "https://github.com/cython/cython/issues/6249#issuecomment-2218231253",
            "id": 2218231253,
            "issue_url": "https://api.github.com/repos/cython/cython/issues/6249",
            "node_id": "IC_kwDOABDGAc6EN4XV",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 1,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 1,
                "url": "https://api.github.com/repos/cython/cython/issues/comments/2218231253/reactions"
            },
            "updated_at": "2024-07-09T16:56:20Z",
            "url": "https://api.github.com/repos/cython/cython/issues/comments/2218231253",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/10536947?v=4",
                "events_url": "https://api.github.com/users/da-woods/events{/privacy}",
                "followers_url": "https://api.github.com/users/da-woods/followers",
                "following_url": "https://api.github.com/users/da-woods/following{/other_user}",
                "gists_url": "https://api.github.com/users/da-woods/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/da-woods",
                "id": 10536947,
                "login": "da-woods",
                "node_id": "MDQ6VXNlcjEwNTM2OTQ3",
                "organizations_url": "https://api.github.com/users/da-woods/orgs",
                "received_events_url": "https://api.github.com/users/da-woods/received_events",
                "repos_url": "https://api.github.com/users/da-woods/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/da-woods/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/da-woods/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/da-woods",
                "user_view_type": "public"
            }
        },
        {
            "author_association": "NONE",
            "body": "I tried running cython normally on the file without setuptools and it built fine, so maybe something is missing/wasn't updated in cythonize?\r\n\r\nEdit:\r\nActually nvm it seems like setuptools also work now for unknown reasons, so I have no idea how I fixed this. I removed numpy from Cython/Includes as a debugging step so maybe that is what fixed it but I'm not really sure. The only other thing that comes to mind is reinstalling setuptools.\r\n\r\n```sh\r\ncython module.pyx\r\ngcc -shared -pthread -fPIC -march=native -fwrapv -O2 -Wall -fno-strict-aliasing -I/usr/include/python3.12 -I/usr/lib/python3.12/site-packages/numpy/_core/include -o module.so module.c\r\n```",
            "created_at": "2024-07-11T20:36:21Z",
            "html_url": "https://github.com/cython/cython/issues/6249#issuecomment-2223888359",
            "id": 2223888359,
            "issue_url": "https://api.github.com/repos/cython/cython/issues/6249",
            "node_id": "IC_kwDOABDGAc6Ejdfn",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/cython/cython/issues/comments/2223888359/reactions"
            },
            "updated_at": "2024-07-11T20:43:06Z",
            "url": "https://api.github.com/repos/cython/cython/issues/comments/2223888359",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/11880815?v=4",
                "events_url": "https://api.github.com/users/ym1234/events{/privacy}",
                "followers_url": "https://api.github.com/users/ym1234/followers",
                "following_url": "https://api.github.com/users/ym1234/following{/other_user}",
                "gists_url": "https://api.github.com/users/ym1234/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/ym1234",
                "id": 11880815,
                "login": "ym1234",
                "node_id": "MDQ6VXNlcjExODgwODE1",
                "organizations_url": "https://api.github.com/users/ym1234/orgs",
                "received_events_url": "https://api.github.com/users/ym1234/received_events",
                "repos_url": "https://api.github.com/users/ym1234/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/ym1234/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/ym1234/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/ym1234",
                "user_view_type": "public"
            }
        },
        {
            "author_association": "NONE",
            "body": "Another reason that may cause this issue is stale cache. See https://github.com/sagemath/sage/issues/39662 for more information.\n\nNote that numpy-maintained version still uses [the same code](https://github.com/numpy/numpy/blob/da0f6f045d179139d0180da937d223be09eece51/numpy/__init__.cython-30.pxd#L808), the only difference is [it declares `subarray` as a property](https://github.com/numpy/numpy/blob/da0f6f045d179139d0180da937d223be09eece51/numpy/__init__.cython-30.pxd#L274-L276). (thus get compiled to a C function call `PyDataType_SUBARRAY` or something)\n",
            "created_at": "2025-03-10T06:47:59Z",
            "html_url": "https://github.com/cython/cython/issues/6249#issuecomment-2709589403",
            "id": 2709589403,
            "issue_url": "https://api.github.com/repos/cython/cython/issues/6249",
            "node_id": "IC_kwDOABDGAc6hgQ2b",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/cython/cython/issues/comments/2709589403/reactions"
            },
            "updated_at": "2025-03-10T06:50:27Z",
            "url": "https://api.github.com/repos/cython/cython/issues/comments/2709589403",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/25191436?v=4",
                "events_url": "https://api.github.com/users/user202729/events{/privacy}",
                "followers_url": "https://api.github.com/users/user202729/followers",
                "following_url": "https://api.github.com/users/user202729/following{/other_user}",
                "gists_url": "https://api.github.com/users/user202729/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/user202729",
                "id": 25191436,
                "login": "user202729",
                "node_id": "MDQ6VXNlcjI1MTkxNDM2",
                "organizations_url": "https://api.github.com/users/user202729/orgs",
                "received_events_url": "https://api.github.com/users/user202729/received_events",
                "repos_url": "https://api.github.com/users/user202729/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/user202729/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/user202729/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/user202729",
                "user_view_type": "public"
            }
        }
    ],
    "comments": 11,
    "comments_url": "https://api.github.com/repos/cython/cython/issues/6249/comments",
    "created_at": "2024-06-18T15:37:59Z",
    "event_data": [
        {
            "actor": {
                "avatar_url": "https://avatars.githubusercontent.com/u/10536947?v=4",
                "events_url": "https://api.github.com/users/da-woods/events{/privacy}",
                "followers_url": "https://api.github.com/users/da-woods/followers",
                "following_url": "https://api.github.com/users/da-woods/following{/other_user}",
                "gists_url": "https://api.github.com/users/da-woods/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/da-woods",
                "id": 10536947,
                "login": "da-woods",
                "node_id": "MDQ6VXNlcjEwNTM2OTQ3",
                "organizations_url": "https://api.github.com/users/da-woods/orgs",
                "received_events_url": "https://api.github.com/users/da-woods/received_events",
                "repos_url": "https://api.github.com/users/da-woods/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/da-woods/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/da-woods/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/da-woods",
                "user_view_type": "public"
            },
            "commit_id": null,
            "commit_url": null,
            "created_at": "2024-06-18T18:37:49Z",
            "event": "mentioned",
            "id": 13205109718,
            "node_id": "MEE_lADOABDGAc6MrTvrzwAAAAMTFfvW",
            "performed_via_github_app": null,
            "url": "https://api.github.com/repos/cython/cython/issues/events/13205109718"
        },
        {
            "actor": {
                "avatar_url": "https://avatars.githubusercontent.com/u/10536947?v=4",
                "events_url": "https://api.github.com/users/da-woods/events{/privacy}",
                "followers_url": "https://api.github.com/users/da-woods/followers",
                "following_url": "https://api.github.com/users/da-woods/following{/other_user}",
                "gists_url": "https://api.github.com/users/da-woods/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/da-woods",
                "id": 10536947,
                "login": "da-woods",
                "node_id": "MDQ6VXNlcjEwNTM2OTQ3",
                "organizations_url": "https://api.github.com/users/da-woods/orgs",
                "received_events_url": "https://api.github.com/users/da-woods/received_events",
                "repos_url": "https://api.github.com/users/da-woods/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/da-woods/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/da-woods/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/da-woods",
                "user_view_type": "public"
            },
            "commit_id": null,
            "commit_url": null,
            "created_at": "2024-06-18T18:37:49Z",
            "event": "subscribed",
            "id": 13205109727,
            "node_id": "SE_lADOABDGAc6MrTvrzwAAAAMTFfvf",
            "performed_via_github_app": null,
            "url": "https://api.github.com/repos/cython/cython/issues/events/13205109727"
        },
        {
            "actor": {
                "avatar_url": "https://avatars.githubusercontent.com/u/11070318?v=4",
                "events_url": "https://api.github.com/users/alekseik1/events{/privacy}",
                "followers_url": "https://api.github.com/users/alekseik1/followers",
                "following_url": "https://api.github.com/users/alekseik1/following{/other_user}",
                "gists_url": "https://api.github.com/users/alekseik1/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/alekseik1",
                "id": 11070318,
                "login": "alekseik1",
                "node_id": "MDQ6VXNlcjExMDcwMzE4",
                "organizations_url": "https://api.github.com/users/alekseik1/orgs",
                "received_events_url": "https://api.github.com/users/alekseik1/received_events",
                "repos_url": "https://api.github.com/users/alekseik1/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/alekseik1/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/alekseik1/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/alekseik1",
                "user_view_type": "public"
            },
            "commit_id": "bd182225719ed4d70f2ba74d0888c4730f116c7a",
            "commit_url": "https://api.github.com/repos/alekseik1/ifg-py/commits/bd182225719ed4d70f2ba74d0888c4730f116c7a",
            "created_at": "2024-07-16T13:08:48Z",
            "event": "referenced",
            "id": 13520708622,
            "node_id": "REFE_lADOABDGAc6MrTvrzwAAAAMl5aQO",
            "performed_via_github_app": null,
            "url": "https://api.github.com/repos/cython/cython/issues/events/13520708622"
        },
        {
            "actor": {
                "avatar_url": "https://avatars.githubusercontent.com/u/120698?v=4",
                "events_url": "https://api.github.com/users/aaschwanden/events{/privacy}",
                "followers_url": "https://api.github.com/users/aaschwanden/followers",
                "following_url": "https://api.github.com/users/aaschwanden/following{/other_user}",
                "gists_url": "https://api.github.com/users/aaschwanden/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/aaschwanden",
                "id": 120698,
                "login": "aaschwanden",
                "node_id": "MDQ6VXNlcjEyMDY5OA==",
                "organizations_url": "https://api.github.com/users/aaschwanden/orgs",
                "received_events_url": "https://api.github.com/users/aaschwanden/received_events",
                "repos_url": "https://api.github.com/users/aaschwanden/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/aaschwanden/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/aaschwanden/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/aaschwanden",
                "user_view_type": "public"
            },
            "commit_id": "8db1f34de3f39109652da57b45c36193e82e4087",
            "commit_url": "https://api.github.com/repos/pism/glacier-flow-tools/commits/8db1f34de3f39109652da57b45c36193e82e4087",
            "created_at": "2024-08-16T00:04:28Z",
            "event": "referenced",
            "id": 13904237438,
            "node_id": "REFE_lADOABDGAc6MrTvrzwAAAAM8wdN-",
            "performed_via_github_app": null,
            "url": "https://api.github.com/repos/cython/cython/issues/events/13904237438"
        },
        {
            "actor": {
                "avatar_url": "https://avatars.githubusercontent.com/u/16732293?v=4",
                "events_url": "https://api.github.com/users/mjziebarth/events{/privacy}",
                "followers_url": "https://api.github.com/users/mjziebarth/followers",
                "following_url": "https://api.github.com/users/mjziebarth/following{/other_user}",
                "gists_url": "https://api.github.com/users/mjziebarth/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/mjziebarth",
                "id": 16732293,
                "login": "mjziebarth",
                "node_id": "MDQ6VXNlcjE2NzMyMjkz",
                "organizations_url": "https://api.github.com/users/mjziebarth/orgs",
                "received_events_url": "https://api.github.com/users/mjziebarth/received_events",
                "repos_url": "https://api.github.com/users/mjziebarth/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/mjziebarth/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/mjziebarth/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/mjziebarth",
                "user_view_type": "public"
            },
            "commit_id": "cbf356c259406178a323e0e27001362d25eda144",
            "commit_url": "https://api.github.com/repos/mjziebarth/SHGOFAST/commits/cbf356c259406178a323e0e27001362d25eda144",
            "created_at": "2024-09-22T19:59:01Z",
            "event": "referenced",
            "id": 14355437098,
            "node_id": "REFE_lADOABDGAc6MrTvrzwAAAANXppYq",
            "performed_via_github_app": null,
            "url": "https://api.github.com/repos/cython/cython/issues/events/14355437098"
        }
    ],
    "events_url": "https://api.github.com/repos/cython/cython/issues/6249/events",
    "html_url": "https://github.com/cython/cython/issues/6249",
    "id": 2360163307,
    "labels": [],
    "labels_url": "https://api.github.com/repos/cython/cython/issues/6249/labels{/name}",
    "locked": false,
    "milestone": null,
    "node_id": "I_kwDOABDGAc6MrTvr",
    "number": 6249,
    "performed_via_github_app": null,
    "reactions": {
        "+1": 0,
        "-1": 0,
        "confused": 0,
        "eyes": 0,
        "heart": 0,
        "hooray": 0,
        "laugh": 0,
        "rocket": 0,
        "total_count": 0,
        "url": "https://api.github.com/repos/cython/cython/issues/6249/reactions"
    },
    "repository_url": "https://api.github.com/repos/cython/cython",
    "state": "open",
    "state_reason": null,
    "sub_issues_summary": {
        "completed": 0,
        "percent_completed": 0,
        "total": 0
    },
    "timeline_url": "https://api.github.com/repos/cython/cython/issues/6249/timeline",
    "title": "[BUG] Latest release (3.0.10) includes NumPy code that won't build with NumPy 2.0",
    "updated_at": "2025-03-10T06:50:27Z",
    "url": "https://api.github.com/repos/cython/cython/issues/6249",
    "user": {
        "avatar_url": "https://avatars.githubusercontent.com/u/520669?v=4",
        "events_url": "https://api.github.com/users/dhermes/events{/privacy}",
        "followers_url": "https://api.github.com/users/dhermes/followers",
        "following_url": "https://api.github.com/users/dhermes/following{/other_user}",
        "gists_url": "https://api.github.com/users/dhermes/gists{/gist_id}",
        "gravatar_id": "",
        "html_url": "https://github.com/dhermes",
        "id": 520669,
        "login": "dhermes",
        "node_id": "MDQ6VXNlcjUyMDY2OQ==",
        "organizations_url": "https://api.github.com/users/dhermes/orgs",
        "received_events_url": "https://api.github.com/users/dhermes/received_events",
        "repos_url": "https://api.github.com/users/dhermes/repos",
        "site_admin": false,
        "starred_url": "https://api.github.com/users/dhermes/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/dhermes/subscriptions",
        "type": "User",
        "url": "https://api.github.com/users/dhermes",
        "user_view_type": "public"
    }
}